packages/react-button/src/components/Button/useButtonBehavior.ts

```diff
@@ -0,0 +1,31 @@
+import { buttonBehavior } from '@fluentui/accessibility';
+import { getCode } from '@fluentui/keyboard-key';
+import { ButtonProps, ButtonState } from './Button.types';
+
+export const useButtonBehavior = (props: ButtonProps): Partial<ButtonState> => {
+  const { as, disabled, loading, onClick, onKeyDown } = props;
+
+  const { attributes, keyActions } = buttonBehavior({ as: as as string, disabled, loading });
```

Comment:
V0 code has the same issue. `as` usage is unsafe to use there as it will not work for cases where a component will be passed. In this case it's preferable to handle it via `ref` so will always know an actual `tagName`:

https://github.com/reakit/reakit/blob/9bd6882fb50948c616e757d750b436decc36ddfc/packages/reakit/src/Button/Button.ts#L40

Reply:
Thanks for this suggestion! I've made the change.